torchdynamo/utils.py

```diff
@@ -65,10 +64,59 @@
 }
 
 
-@functools.lru_cache(None)
 def init_logging():
     if "PYTEST_CURRENT_TEST" not in os.environ:
         logging.config.dictConfig(LOGGING_CONFIG)
+        logger = logging.getLogger(config.log_name)
+        logger.setLevel(config.log_level)
+
+
+def get_logger():
+    return logging.getLogger(config.log_name)
+
+
+def log_warning(msg, *args, **kwargs):
+    get_logger().warning(msg, *args, **kwargs)
+
+
+def log_error(msg, *args, **kwargs):
+    get_logger().error(msg, *args, **kwargs)
+
+
+def log_info(msg, *args, **kwargs):
+    get_logger().info(msg, *args, **kwargs)
+
+
+def log_debug(msg, *args, **kwargs):
+    get_logger().debug(msg, *args, **kwargs)
+
+
+# filter out all frames after entering dynamo
+def filter_stack(stack):
+    user_stack = []
+    for frame in stack:
+        if "convert_frame" in frame.filename:
+            break
+        if "eval_frame" in frame.filename:
+            continue
+        user_stack.append(frame)
+
+    return user_stack
+
+
+def format_graph_tabular(graph):
```

Comment:
when would I use this vs `graph.print_tabular()`?

Reply:
`print_tabular()` only prints unfortunately and doesn't return the string

Reply:
Right - but we can make it either `print_tabular(file)` where file is a stream, or:

```
def print_tabular():
   print(tabular_out(self.graph))
```

And we use `fx.tabular_out(graph)` - if that make sense?

Reply:
Since we're changing this code can we also include the line numbers associated in the python file to the lines in the fx graph

Reply:
@SherlockNoMad already put stack_trace on fx nodes.

Reply:
Yeah I'll upstream the refactor to fx, I forgot we own that too :D

Reply:
I'll upstream this and remove once we have the version of Pytorch that contains the update